private/pkg/storage/diff.go

```diff
@@ -78,6 +78,20 @@ func DiffWithExternalPathPrefixes(
 	}
 }
 
+// DiffWithTransform returns a DiffOption that adds a transform function. The transform function will be run on each
+// file being compared before it is diffed. transform takes the arguments:
+//  side: one or two whether it is the first or second item in the diff
+//  filename: the filename including path
+//  content: the file content.
+// transform returns a string that is the transformed content of filename.
+func DiffWithTransform(
```

Comment:
It'd be great if we could implement this functionality in `storagetesting` since it's only relevant for test functions. But given the nature of how you need to apply this option in the `Diff` implementation itself, I'm not sure it's possible.

Reply:
I poked around at it for a minute, but like you said there isn't a good way to move this to another package without significant changes. We could move all of `diff.go` to `storagetesting`, but it is also used by `./private/pkg/storage/cmd/ddiff`. It doesn't seem right to have a `main` package depend on a package with "testing" in the name.